src/borg/archiver.py

```diff
@@ -433,6 +435,14 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = 0
+            for item in archive.iter_items(filter=lambda item: matcher.match(item.path)):
+                extracted_size += item.file_size()
```

Comment:
hmm, as it is only telling the percentage of extracted size, yes, that is a known value and thus correct.

the whole thing is still questionable though as people will try to correlate percentage with time ("it took 1h for 10%, so it will be finished in 9h") and the speed will vary widely.